pyproject.toml

```
@@ -75,7 +75,7 @@ pytest-operator = "^0.35.0"
 pytest-operator-cache = {git = "https://github.com/canonical/data-platform-workflows", tag = "v16.7.0", subdirectory = "python/pytest_plugins/pytest_operator_cache"}
 pytest-operator-groups = {git = "https://github.com/canonical/data-platform-workflows", tag = "v16.7.0", subdirectory = "python/pytest_plugins/pytest_operator_groups"}
 pytest-microceph = {git = "https://github.com/canonical/data-platform-workflows", tag = "v16.7.0", subdirectory = "python/pytest_plugins/microceph"}
-juju = "^3.4.0.0"
+juju = "3.4.0.0"
```

```suggestion
juju = "~3.4.0"
```

This way we guarantee that the dependency bump only happens within `3.4.x` series